src/ray/core_worker/experimental_mutable_object_manager.cc

```diff
@@ -296,8 +298,8 @@ Status MutableObjectManager::ReadAcquire(const ObjectID &object_id,
   channel->reading = true;
 
   int64_t version_read = 0;
-  Status s =
-      object->header->ReadAcquire(sem, channel->next_version_to_read, version_read);
+  Status s = object->header->ReadAcquire(
```

Comment:
We also need to timeout the lock acquire at this [line](https://github.com/ray-project/ray/blob/496e0c4e039c3c3fb505cd6d419955bb6453c1ab/src/ray/core_worker/experimental_mutable_object_manager.cc#L297).